pkg/repository/v1manifest/repo.go

```diff
@@ -318,7 +290,7 @@ func (manifest *Timestamp) SetSnapshot(s *Manifest) (*Timestamp, error) {
 }
 
 // SetRole populates role list in the root manifest
-func (manifest *Root) SetRole(m ValidManifest) {
+func (manifest *Root) SetRole(m ValidManifest) error {
```

Comment:
What did the `error` represent? It seems this function doesn't produce any error at all.